docs/developing_recipes_locally.md

```diff
@@ -0,0 +1,207 @@
+For ConanCenter Hook errors, go to the [Error Knowledge Base](error_knowledge_base.md) page to know more about those.
+
+To test with the same environment, the [build images](supported_platforms_and_configurations.md#build-images) are available.
+Instructions for using these images can be found in [Testing more environments](#testing-more-environments)
+
+## Running the Python Linters
+
+Linters are always executed by Github actions to validate parts of your recipe, for instance, if it uses migrated Conan tools imports.
+To understand how to run linters locally, read [V2 linter](v2_linter.md) documentation.
+
+It is possible to run the linter locally the same way it is being run [using Github actions](../.github/workflows/linter-conan-v2.yml):
+
+ * (Recommended) Use a dedicated Python virtualenv.
+ * Ensure you have required tools installed: `conan` and `pylint` (better to uses fixed versions)
+
+   ```
+   pip install conan~=1.0 pylint==2.14
+   ```
+
+ * Set environment variable `PYTHONPATH` to the root of the repository
+
+   ```
+   export PYTHONPATH=your/path/conan-center-index
+   ```
+
+  * Now you just need to execute the `pylint` commands:
+
+    ```
+    # Lint a recipe:
+    pylint --rcfile=linter/pylintrc_recipe recipes/boost/all/conanfile.py
+
+    # Lint the test_package
+    pylint --rcfile=linter/pylintrc_testpackage recipes/boost/all/test_package/conanfile.py
+    ```
+
+## Testing the different `test_*_package`
+
+This can be selected when calling `conan create` or sperately with `conan test`
+
+```sh
+conan create recipes/fmt/all/conanfile.py 9.0.0@ -tf test_v1_package/ -pr:b=default -pr:h=default
+```
+
+```sh
+conan test recipes/fmt/all/test_v1_package/conanfile.py fmt/9.0.0@ -pr:h=default -pr:b=default
+```
+
+## Testing more environments
+
+This can be difficult for some platforms given virtualization support.
+
+For Windows and MacOS users, you can test the Linux build environments by with the the Docker build images.
+
+Assuming you've already tested it locally, and it's successfully export to your cache, you can:
+
+1. Creating a new profile.
+2. Build missing packages
+
+Example.
+
+```
+docker run -v/Users/barbarian/.conan:/home/conan/.conan conanio/gcc8 bash -c "conan profile new --detect gcc8"
+docker run -v/Users/barbarian/.conan:/home/conan/.conan conanio/gcc8 bash -c "conan install -pr gcc8 fmt/9.0.0@ --build missing"
+```
+
+> **Note**: If you are running on Mac M1, the follow Docker argument is required: `--platform=linux/amd64
+
+## Using Conan 2.0
+
+Everything you need to know about the methods, commands line, outputs can be found in the
+[Conan 2.0 Migrations](https://docs.conan.io/en/latest/conan_v2.html) docs. This should be fairly straight forward.
+Conan 2.0 by default has a different `CONAN_USER_HOME` location. You can have to separate caches, profiles
+and settings. So no need to worry about breaking your v1 setup.
```

Comment:
```suggestion
Conan 2.0 by default has a different `CONAN_USER_HOME` location, which means that it has separate 
caches, profiles, and settings.
This will leave your Conan 1.0 setup completely intact when using Conan 2.0.
```